src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3510,7 +3513,7 @@ export class StudyViewPageStore
                 const uniqueKey = item.attributeId;
                 if (this.isNewlyAdded(uniqueKey)) {
                     this.showAsPieChart(uniqueKey, item.counts.length);
-                    this.newlyAddedCharts.remove(uniqueKey);
+                    //this.newlyAddedCharts.remove(uniqueKey);
```

Comment:
commenting this out seems to fix problem.  i think that mutating this observable state here causes a promise to reinvoke and resolve with empty [].  leading to blank chart

Reply:
Perhaps newlyAddedCharts doesn't really need to be observable?